activemodel/lib/active_model/type/collection.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+module ActiveModel
+  module Type
+    # Attribute type for a collection of values. It is registered under
+    # the +:collection+ key. +:element_type+ option is used to specify elements type
+    #
+    #   class User
+    #     include ActiveModel::Attributes
+    #
+    #     attribute :lucky_numbers, :collection, element_type: :integer
+    #   end
+    #
+    #   user = User.new(lucky_numbers: [1, 2, 3])
+    #   user.lucky_numbers # => [1, 2, 3]
+    #
+    # Value is wrapped into an Array if not an Array already
+    #
+    #   User.new(lucky_numbers: 1).lucky_numbers # => [1]
+    #
+    # Collection elements are coerced by their +:element_type+ type
+    #
+    #   User.new(lucky_numbers: ["1"]).lucky_numbers # => [1]
+    #
+    class Collection < Value
+      def initialize(**args)
+        @element_type = args.delete(:element_type)
+        @type_object = Type.lookup(element_type)
```

Comment:
Can we use the remaining `**args` to instantiate the `@type_object` ?
```suggestion
        @type_object = Type.lookup(element_type, **args)
```
So that we can use the `:collection` type with the `:model` type like so:
```ruby
attribute :things, :collection, element_type: :model, class_name: "Thing"
```

Reply:
It might also be necessary to use `super()` instead of `super` to prevent `unknown keyword` errors.

Reply:
We definitely can! Going to make a change, thanks! I agree that we should probably call `super()` as there is not much we can initialize in `Value` that could be useful for `Model` type
https://github.com/rails/rails/blob/07a5584999240642454ee8fae094ca49895e71b0/activemodel/lib/active_model/type/value.rb#L14

I also wanted to mention that this is a very basic implementation and there is a high chance that we are going to extend capabilities of the type in separate PRs. For example I think that a macro like:
```ruby
class Post
  include ActiveModel::Model
  include ActiveModel::Attributes
  # perhaps class name could be inferred from the attribute name
  embeds_many :comments, class_name: "Comment" 
end
```
is a really good idea so I'd expect more changes related to these types being made on top of this PR

Reply:
In fact, this was my original intention, as I'm seeking to update this PR #43399.